v0/destinations/pipedrive/transform.js

```diff
@@ -0,0 +1,570 @@
+/* eslint-disable no-nested-ternary */
+/* eslint-disable prefer-const */
+/* eslint-disable prettier/prettier */
+/* eslint-disable camelcase */
+const get = require("get-value");
+const set = require("set-value");
+const { EventType } = require("../../../constants");
+const {
+  constructPayload,
+  extractCustomFields,
+  defaultRequestConfig,
+  removeUndefinedAndNullValues,
+  defaultPostRequestConfig,
+  defaultPutRequestConfig,
+  getValueFromMessage,
+  getFieldValueFromMessage,
+  getDestinationExternalID,
+  getErrorRespEvents,
+  getSuccessRespEvents,
+} = require("../../util");
+const {
+  getMergeEndpoint,
+  MAPPING_CONFIG,
+  CONFIG_CATEGORIES,
+  PERSONS_ENDPOINT,
+  PIPEDRIVE_IDENTIFY_EXCLUSION,
+  PIPEDRIVE_GROUP_EXCLUSION,
+  PIPEDRIVE_TRACK_EXCLUSION,
+  LEADS_ENDPOINT,
+  FLATTEN_KEYS
+} = require("./config");
+const {
+  createNewOrganisation,
+  searchPersonByCustomId,
+  searchOrganisationByCustomId,
+  getFieldValueOrThrowError,
+  updateOrganisationTraits,
+  renameCustomFields,
+  getUserIDorExternalID,
+  createPerson,
+  CustomError
+} = require("./util");
+
+const identifyResponseBuilder = async (message, category, { Config }) => {
+  // name is required field for destination payload
+  
+  // destUserId is external Id or provided userId
+  let destUserId = getDestinationExternalID(message, "person_id");
+
+  if(!destUserId) {
+    if (!get(Config, "userIdToken")) {
+      throw new CustomError("userId Token is required", 400);
+    }
+
+    const userId = getFieldValueFromMessage(message, "userIdOnly");
+    if(!userId) {
+      throw new CustomError("userId or person_id required", 400);
+    }
+    destUserId = userId;
```

Comment:
user id will be the custom made field while dest user id is the internal id right?